compiler/rustc_hir/src/tests.rs

```diff
@@ -14,7 +14,7 @@ fn def_path_hash_depends_on_crate_id() {
     // the crate by changing the crate disambiguator (e.g. via bumping the
     // crate's version number).
 
-    create_session_if_not_set_then(Edition::Edition2024, |_| {
+    create_session_globals_then(Edition::Edition2024, || {
```

Comment:
wouldn't this panic if a second test was added?

Reply:
Only if a `create_session_globals_then` gets nested within another one.

You can see test files like `compiler/rustc_expand/src/parse/tests.rs` have multiple tests, each one containing a `create_default_session_globals_then` call.

Reply:
I don't much like the `if_not_set_then` functions for this reason. They only exist because there are some places where initialization is messy and there are code paths that are sometimes executed within a `create_session_globals_then` and sometimes not, depending on whether it's rustc vs. rustdoc. vs. whatever else. They could probably be removed with some effort.

Reply:
I see.